code/modules/clothing/shoes/clownshoes.dm

```diff
@@ -0,0 +1,207 @@
+		return
+	if (!enabled_waddle)
+		to_chat(user, "<span class='notice'>You switch off the waddle dampeners!</span>")
+		enabled_waddle = TRUE
+	else
+		to_chat(user, "<span class='notice'>You switch on the waddle dampeners!</span>")
+		enabled_waddle = FALSE
+
+
+
+/obj/item/clothing/shoes/clown_shoes/sec
+	desc = "The law *IS* a joking matter!"
+	name = "clown jacks"
+	icon_state = "clown_sec"
+	inhand_icon_state = "clown_shoes_sec"
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes
+
+/obj/item/clothing/shoes/clown_shoes/sec/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/footstep/wood5.ogg'=1), 100, 100, 0)
+
+
+/obj/item/clothing/shoes/clown_shoes/prison
+	desc = "That trial was a joke! A kangaroo court!"
+	name = "orange clown shoes"
+	icon_state = "clown_prison"
+	inhand_icon_state = "clown_shoes_prison"
+	slowdown = 15
+
+/obj/item/clothing/shoes/clown_shoes/prison/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/chain.ogg'=1), 50)
+
+
+/obj/item/clothing/shoes/clown_shoes/prison/allow_attack_hand_drop(mob/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/C = user
+		if(C.shoes == src)
+			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
+			return FALSE
+	return ..()
+
+/obj/item/clothing/shoes/clown_shoes/prison/MouseDrop(atom/over)
+	var/mob/m = usr
+	if(ishuman(m))
+		var/mob/living/carbon/human/c = m
+		if(c.shoes == src)
+			to_chat(c, "<span class='warning'>You need help taking these off!</span>")
+			return
+	return ..()
+
+
+/obj/item/clothing/shoes/clown_shoes/engi
+	desc = "These boots were made for working, but that's not what we're gonna do."
+	name = "clown boots"
+	icon_state = "clown_engi"
+	inhand_icon_state = "clown_shoes_engi"
+
+/obj/item/clothing/shoes/clown_shoes/engi/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/footstep/catwalk3.ogg'=1,'sound/effects/footstep/carpet5.ogg'=1), 100)
+
+
+/obj/item/clothing/shoes/clown_shoes/hydro
+	desc = "A bit prickly, and somewhat loose. Perhaps you can get a new sole?"
+	name = "clown grass shoes"
+	icon_state = "clown_hydro"
+	inhand_icon_state = "clown_shoes_hydro"
+
+/obj/item/clothing/shoes/clown_shoes/hydro/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/footstep/grass2.ogg'=1,'sound/effects/footstep/grass3.ogg'=1), 100)
+
+
+/obj/item/clothing/shoes/clown_shoes/clogs
+	desc = "Great sole support, but shouldn't these bee yellow?"
+	name = "clown clogs"
+	icon_state = "clown_clogs"
+	inhand_icon_state = "clown_shoes_clogs"
+
+/obj/item/clothing/shoes/clown_shoes/clogs/attackby(obj/H, mob/user, loc)
+	..()
+	if (H.type == /obj/item/reagent_containers/honeycomb || H.type== /obj/item/grown/sunflower)
+		var/location = get_turf(user)
+		new /obj/item/clothing/shoes/clown_shoes/dutch(location)
+		qdel(H)
+		qdel(src)
+	return
```

Comment:
```suggestion
/obj/item/clothing/shoes/clown_shoes/clogs/attackby(obj/item/attacking_item, mob/user, loc)
	. = ..()
	if (attacking_item.type == /obj/item/reagent_containers/honeycomb || attacking_item.type== /obj/item/grown/sunflower)
		var/location = get_turf(user)
		new /obj/item/clothing/shoes/clown_shoes/dutch(location)
		qdel(attacking_item)
		qdel(src)

```
* Preserve the return.
* Proper type casting on the input.
* Better var names.
* No need to return at the end.